File: pluralscan-core/src/pluralscan/domain/issues/issue_location.py
from dataclasses import dataclass
from enum import Enum
from typing import Optional


class IssueLocationKind(Enum):
    """IssueLocationKind"""

    UNKNOWN = 0
    SOURCE = 1
    METADATA = 2


@dataclass(frozen=True)
class IssueLocation:
    """IssueLocation"""

    path: Optional[str] = None
    absolute_path: Optional[str] = None
    line: int = 0
    column: int = 0
    end_line: int = 0
    end_column: int = 0
    kind: IssueLocationKind = IssueLocationKind.UNKNOWN

File: pluralscan-api-client/src/models/IssueLocation.ts
import { IssueLocationKind } from "./IssueLocationKind";

export interface IssueLocation {
    path: string;
    absolute_path: string;
    line: number;
    column: number;
    end_line: number;
    end_column: number;
    kind: IssueLocationKind
}